stan/math/prim/scal/err/throw_domain_error_vec.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef STAN_MATH_PRIM_SCAL_ERR_THROW_DOMAIN_ERROR_VEC_HPP
+#define STAN_MATH_PRIM_SCAL_ERR_THROW_DOMAIN_ERROR_VEC_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/scal/err/throw_domain_error.hpp>
+#include <sstream>
+#include <string>
+
+namespace stan {
+namespace math {
+
+/**
+ * Throw a domain error with a consistently formatted message.
+ * This is an abstraction for all Stan functions to use when throwing
+ * domain errors. This will allow us to change the behavior for all
+ * functions at once. (We've already changed behavior mulitple times up
```

Comment:
We don't need to include this kind of historical note---it's enough to say we should use this abstraction to control throwing behavior centrally.

Reply:
Also this was preexisting, but I'm happy to remove it so the docs match those for `throw_domain_error`.